eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/SortClauseHandler/Random/PgSqlRandom.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\SortClauseHandler\Random;
+
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\SortClauseHandler\AbstractRandom;
+
+class PgSqlRandom extends AbstractRandom
+{
+    public function applyJoin(
+        SelectQuery $query,
+        SortClause $sortClause,
+        $number,
+        array $languageSettings
+    ) {
+        $seedAsInt = $sortClause->targetData->seed;
+        if (!$seedAsInt) {
+            return;
+        }
+
+        $seed = $this->interpolateIntegerToFloatRange($seedAsInt);
+        $query
+            ->innerJoin(
+                '(select setseed(:seed)) as seed',
+                0,
+                0
```

Comment:
It looks very artificial to me. What is the query produced by this? I tried to debug it, but failed :(

Reply:
This is equivalent to `CROSS JOIN`.
So this is `INNER JOIN (select setseed(:seed)) as seed ON 0 = 0`

Reply:
AFAIS all examples showing `SELECT setseed(:seed);` to be executed before the actual query with `random()` or using `SELECT in WITH` (https://www.postgresql.org/docs/9.4/queries-with.html) queries e.g.:

```sql
WITH seed AS (SELECT setseed(.123)) 
SELECT random();
```

But both ways seems to be hard to implement 🤔

Reply:
Hmm, seems this might be possible only if we rewrite entire query building for LSE to Doctrine.